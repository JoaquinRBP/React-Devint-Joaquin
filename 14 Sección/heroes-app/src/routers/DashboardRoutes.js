import React from 'react'
import { Navbar } from '../components/ui/Navbar';
import { Routes, Route, Link } from "react-router-dom";
import { MarvelScreen } from '../components/marvel/MarvelScreen';
import { HeroScreen } from '../components/heroes/HeroScreen';
import { DcScreen } from '../components/dc/DcScreen';
import { SearchScreen } from '../components/search/SearchScreen';

export const DashboardRoutes = () => {
return(
    <>
        <Navbar />
        <div className="container mt-2">
            <Routes>
                <Route exact path="/marvel" element={<MarvelScreen />}/>
                <Route exact path="/hero/:heroId/*" element={<HeroScreen />}/>
                <Route exact path="/dc" element={<DcScreen />}/>
                <Route exact path="/search" element={<SearchScreen />}/>
                <Route path='*' element={<main>
                    <h1>ERR 404</h1>
                </main>} />
            </Routes>
        </div>
    </>
)
}
