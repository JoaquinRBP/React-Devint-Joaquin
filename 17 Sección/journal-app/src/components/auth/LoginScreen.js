import React from 'react'
import { Link } from 'react-router-dom'
import {useForm} from '../../hooks/useForm';
import {useDispatch} from 'react-redux';
import validator from 'validator';
import { startGoogleLogin, startLoginEmailPassword } from '../../actions/auth';
import { EvalError, removeError } from '../../actions/ui';
import { useSelector } from 'react-redux';
export const LoginScreen = () => {
    //useDispatch -- hook
    const dispatch =useDispatch();
    const [values, handleInputChange] = useForm({
        email:'joaquin@gmail.com',
        password:123456
    });
    const {loading} = useSelector( state => state.ui);
    const {email, password} = values;
    const handleLogin = (e) => {
        e.preventDefault();
        if(isFormValid()){
            dispatch(startLoginEmailPassword(email,password));
        }
    }
    const handleGoogleLogin = (e) => {
        e.preventDefault();
        dispatch(startGoogleLogin());
    }
    const isFormValid = () => {
        if(!validator.isEmail(email)){
            dispatch(EvalError('Email is not valid'));
            //console.log('');
            return false;
        }else if(password.length<5){
            dispatch(EvalError('Password is not valid'));
            //console.log('Password is not valid');
            return false;
        }
            dispatch(removeError());
        return true;
    }
return (
    <>
        <h3 className="auth__title">LoginScreen</h3>
        <form 
        className="animate__animated animate__fadeIn animate__faster"
        onSubmit={handleLogin}>
            <input 
                type="text"
                placeholder="Email"
                name="email"
                className="auth__input"
                value={email}
                onChange={handleInputChange}
            />
            <input 
                type="pasword"
                placeholder="Password"
                name="password"
                className="auth__input"
                autoComplete="off"
                value={password}
                onChange={handleInputChange}
            />
            <button 
                type="submit"
                className="btn btn-primary btn-block"
                disabled={loading}
            >
                Login
            </button>
            <div className="auth__social-networks">
                <p>Login with Social Networks</p>
                    <div 
                        className="google-btn"
                        onClick={handleGoogleLogin}
                    >
                    <div className="google-icon-wrapper">
                        <img className="google-icon" src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg" alt="google button" />
                    </div>
                    <p className="btn-text">
                        <b>Sign in with google</b>
                    </p>
                    </div>
            </div>
            <Link className="link" to="/auth/register">
                Create a new account
            </Link>
        </form>
    </>
)
}
