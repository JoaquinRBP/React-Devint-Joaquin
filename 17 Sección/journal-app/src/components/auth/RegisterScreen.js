import React from 'react'
import { Link } from 'react-router-dom'

export const RegisterScreen = () => {
return (
    <>
        <h3 className="auth__title">LoginScreen</h3>
        <form>
            <input 
                type="text"
                placeholder="Name"
                name="name"
                className="auth__input"
            />
            <input 
                type="text"
                placeholder="Email"
                name="email"
                className="auth__input"
            />
            <input 
                type="pasword"
                placeholder="Password"
                name="password"
                className="auth__input"
                autoComplete="off"
            />
            <input 
                type="pasword"
                placeholder="Confirm"
                name="password2"
                className="auth__input"
                autoComplete="off"
            />
            <button 
                type="submit"
                className="btn btn-primary btn-block mb-5"
            >
                Register
            </button>
            <Link className="link mt-5" to="/auth/login">
                All Ready Register?
            </Link>
        </form>
    </>
)
}
