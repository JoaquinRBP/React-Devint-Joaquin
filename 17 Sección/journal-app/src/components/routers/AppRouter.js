import React from 'react'
import { BrowserRouter as Router, Navigate } from "react-router-dom";
import { Routes, Route, Link} from "react-router-dom";
import { JournalScreen } from '../journal/JournalScreen';
import { AuthRouter } from './AuthRouter';
export const AppRouter = () => {
return (
    <Router>
        <div>
        <Routes>
            <Route path="/*" element={<AuthRouter />} />
            <Route exact path="/" element={<JournalScreen />} />
        </Routes>
        </div>
    </Router>
)
}
