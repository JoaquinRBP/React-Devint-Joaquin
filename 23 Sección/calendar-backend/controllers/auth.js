const {response} = require('express');
const { validationResult} = require('express-validator');
const crearUsuario = (req,res = response) => {
    
    const {nombre, email, password} = req.body;
    // if(nombre.length <5) {
    //     return res.status(400).json({
    //         ok: false,
    //         msg:'El nombre debe de ser mayor o igual a 5 letras'
    //     })
    // }

    res.status(201).json(
        {
            ok: true,
            msg:'register',
            nombre,
            email,
            password
        }
        ); 
}
const loginUsuario = (req,res=response) => {
    const {email, password} = req.body
    
    
    res.json({
        ok: true,
        msg:'Login',
        email,
        password
    });
}
const revalidarToken = (req,res=response) => {
    res.json({
        ok: true,
        msg: 'renew'
    });
}
module.exports = {
    crearUsuario,
    loginUsuario,
    revalidadToken: revalidarToken,
};