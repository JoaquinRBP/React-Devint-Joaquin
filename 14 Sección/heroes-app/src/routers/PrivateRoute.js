import React from 'react'
import { Navigate, useLocation} from 'react-router';
import {PropTypes} from 'prop-types';
import { DashboardRoutes } from './DashboardRoutes';

export const PrivateRoute = ({
    isAuthenticated,
}) => {
    const {pathname} = useLocation();
    localStorage.setItem('lastpath',pathname);
return isAuthenticated? <DashboardRoutes />: <Navigate to='/login'/>;
}

PrivateRoute.propTypes = {
    isAuthenticated: PropTypes.bool.isRequired,
}
