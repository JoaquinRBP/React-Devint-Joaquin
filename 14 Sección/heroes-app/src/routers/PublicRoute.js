import React from 'react'
import {LoginScreen} from '../components/login/LoginScreen';
import {PropTypes} from 'prop-types';
import { DashboardRoutes } from './DashboardRoutes';
import { Navigate } from 'react-router';

export const PublicRoute = ({
    isAuthenticated,
}) => {
return !isAuthenticated?<LoginScreen />: <Navigate to='/' />;
}

PublicRoute.propTypes = {
    isAuthenticated: PropTypes.bool.isRequired,
}

