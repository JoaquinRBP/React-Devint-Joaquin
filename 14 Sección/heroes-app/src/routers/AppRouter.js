import React, { useContext } from 'react'
import { BrowserRouter as Router } from "react-router-dom";
import { Routes, Route, Link} from "react-router-dom";
import { AuthContext } from '../auth/AuthContext';
import {PrivateRoute} from './PrivateRoute';
import { PublicRoute } from './PublicRoute';
export const AppRouter = () => {
    const {user} = useContext(AuthContext);
return (
    <Router>
        <div>
        <Routes>
            <Route exact path="/login" element={<PublicRoute isAuthenticated={user.logged} />}/>
            <Route path="*" element={<PrivateRoute isAuthenticated={user.logged}/>}/>
        </Routes>
        </div>
    </Router>
)
}
