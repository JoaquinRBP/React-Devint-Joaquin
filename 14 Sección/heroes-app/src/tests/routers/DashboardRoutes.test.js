import {AuthContext} from '../../auth/AuthContext';
import {DashboardRoutes} from '../../routers/DashboardRoutes';
import React from 'react';
import {shallow, mount} from 'enzyme';
import { MemoryRouter } from 'react-router';
jest.mock('react-router',() => ({
    ...jest.requireActual('react-router'),
    useNavigate: ()=> jest.fn()
}));
describe('Prueba en DashboardRoutes', () => {
    const contextValue ={
        dispatch: jest.fn(),
        user:{
            logged:true,
            name: 'Joaquin'
        }
    }
    test('Debe mostrarse correctamente', () => {
        const wrapper = mount(
            <MemoryRouter>
            <AuthContext.Provider value={contextValue}>
                <DashboardRoutes />
            </AuthContext.Provider>
            </MemoryRouter>
        );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.text-info').text().trim()).toBe('Joaquin');
    })
    
})
