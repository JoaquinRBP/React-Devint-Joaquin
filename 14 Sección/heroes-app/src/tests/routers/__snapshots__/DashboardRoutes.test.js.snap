// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en DashboardRoutes Debe mostrarse correctamente 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      14 Secci√≥n
    </a>
    <div
      className="collapse navbar-collapse"
      id="navbarNavAltMarkup"
    >
      <div
        className="navbar-nav"
      >
        <a
          activeclassname="active"
          aria-current="page"
          className="nav-item nav-link active"
          exact="true"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
        <a
          activeclassname="active"
          className="nav-item nav-link"
          exact="true"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          activeclassname="active"
          className="nav-item nav-link"
          exact="true"
          href="/dc"
          onClick={[Function]}
        >
          Dc
        </a>
        <a
          activeclassname="active"
          className="nav-item nav-link"
          exact="true"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <p
          className="nav-item nav-link text-info"
        >
          Joaquin
        </p>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container mt-2"
  >
    <main>
      <h1>
        ERR 404
      </h1>
    </main>
  </div>,
]
`;
